src/game/option_graphics.c

```diff
@@ -26,156 +35,283 @@ typedef enum GRAPHICS_TEXT {
+                Text_Create(x, y, g_GameFlow.strings[col->text]);
+            Text_CentreV(m_Text[col->option], 1);
+        }
 
-    sprintf(
-        buf, g_GameFlow.strings[GS_DETAIL_VSYNC_FMT],
-        g_GameFlow.strings
-            [g_Config.rendering.enable_vsync ? GS_MISC_ON : GS_MISC_OFF]);
-    m_Text[TEXT_VSYNC] = Text_Create(0, y, buf);
-    y += ROW_HEIGHT;
+        ys[col->col_num] += ROW_HEIGHT;
+        max_y = MAX(max_y, ys[col->col_num]);
+    }
 
-    sprintf(
-        buf, g_GameFlow.strings[GS_DETAIL_BRIGHTNESS_FMT], g_Config.brightness);
-    m_Text[TEXT_BRIGHTNESS] = Text_Create(0, y, buf);
-    y += ROW_HEIGHT;
+    sprintf(buf, g_GameFlow.strings[GS_DETAIL_FLOAT_FMT], g_Config.brightness);
+    Text_ChangeText(m_Text[TEXT_BRIGHTNESS_TOGGLE], buf);
 
     sprintf(
-        buf, g_GameFlow.strings[GS_DETAIL_UI_TEXT_SCALE_FMT],
-        g_Config.ui.text_scale);
-    m_Text[TEXT_UI_TEXT_SCALE] = Text_Create(0, y, buf);
-    y += ROW_HEIGHT;
+        buf, g_GameFlow.strings[GS_DETAIL_FLOAT_FMT], g_Config.ui.text_scale);
+    Text_ChangeText(m_Text[TEXT_UI_TEXT_SCALE_TOGGLE], buf);
 
     sprintf(
-        buf, g_GameFlow.strings[GS_DETAIL_UI_BAR_SCALE_FMT],
-        g_Config.ui.bar_scale);
-    m_Text[TEXT_UI_BAR_SCALE] = Text_Create(0, y, buf);
-    y += ROW_HEIGHT;
+        buf, g_GameFlow.strings[GS_DETAIL_FLOAT_FMT], g_Config.ui.bar_scale);
+    Text_ChangeText(m_Text[TEXT_UI_BAR_SCALE_TOGGLE], buf);
 
-    char tmp[10];
     sprintf(
-        tmp, "%dx%d", Screen_GetPendingResWidth(),
-        Screen_GetPendingResHeight());
-    sprintf(buf, g_GameFlow.strings[GS_DETAIL_VIDEO_MODE_FMT], tmp);
-    m_Text[TEXT_RESOLUTION] = Text_Create(0, y, buf);
-    y += ROW_HEIGHT;
-
-    if (g_OptionSelected < TEXT_OPTION_MIN) {
-        g_OptionSelected = TEXT_OPTION_MIN;
-    }
-    if (g_OptionSelected > TEXT_OPTION_MAX) {
-        g_OptionSelected = TEXT_OPTION_MAX;
-    }
+        buf, g_GameFlow.strings[GS_DETAIL_RESOLUTION_FMT],
+        Screen_GetPendingResWidth(), Screen_GetPendingResHeight());
+    Text_ChangeText(m_Text[TEXT_RESOLUTION_TOGGLE], buf);
 
-    Text_AddBackground(m_Text[TEXT_TITLE_BORDER], ROW_WIDTH, y - TOP_Y, 0, 0);
+    int16_t width = ROW_WIDTH;
+    int16_t height = max_y + BORDER * 2 - TOP_Y;
+    Text_AddBackground(m_Text[TEXT_TITLE_BORDER], width, height, 0, 0);
     Text_AddOutline(m_Text[TEXT_TITLE_BORDER], 1);
 
-    Text_AddBackground(m_Text[TEXT_TITLE], ROW_WIDTH - 4, 0, 0, 0);
-    Text_AddOutline(m_Text[TEXT_TITLE], 1);
-
-    Text_AddBackground(m_Text[g_OptionSelected], ROW_WIDTH - 12, 0, 0, 0);
-    Text_AddOutline(m_Text[g_OptionSelected], 1);
+    m_Text[TEXT_LEFT_ARROW] = Text_Create(0, 0, "@");
+    Text_CentreV(m_Text[TEXT_LEFT_ARROW], 1);
+    Text_SetPos(
+        m_Text[TEXT_LEFT_ARROW], m_Text[TEXT_PERSPECTIVE_TOGGLE]->pos.x - 20,
+        m_Text[TEXT_PERSPECTIVE_TOGGLE]->pos.y);
+    m_HideLeft = g_Config.rendering.enable_perspective_filter ? false : true;
```

Comment:
We generally avoid idioms like `bool_var ? true : false`, just use `bool_var`. In this case `!bool_var`.